### PR TITLE
Refactor implicit conversions to VDomModifier

### PR DESCRIPTION
**Pull request by mariusmuja**
More consistent handling of implicit conversions to `VDomModifier` though the `AsVDomModifier` type class. This removes all (except one) implicit conversions to `VDomModifier` from the `dom` package object.

It also allows to have `Seq[String]`, `Seq[Int]` (any `Seq[T : AsVomModifier]`) in the VDom (see #155).

### CODE REVIEW

File: src/test/scala/outwatch/OutWatchDomSpec.scala [hunk 1]
```diff
@@ -824,6 +824,29 @@ class OutWatchDomSpec extends JSDomSpec {
...
+    val node = document.createElement("div")
+
+    OutWatch.renderReplace(node, div("one")).unsafeRunSync()
+    node.innerHTML shouldBe "one"
+
+    OutWatch.renderReplace(node, div(1)).unsafeRunSync()
+    node.innerHTML shouldBe "1"
+
+    OutWatch.renderReplace(node, div(1.0)).unsafeRunSync()
+    node.innerHTML shouldBe "1"
```
Comment (fdietze) [inline src/test/scala/outwatch/OutWatchDomSpec.scala]: Why is a `Double` `1.0` rendered as `1` and not `1.0`?

Comment (mariusmuja) [reply src/test/scala/outwatch/OutWatchDomSpec.scala]: Apparently that's how Scala.js renders `Double.toString`:

https://scalafiddle.io/sf/OcJkymj/0

Comment (fdietze) [reply src/test/scala/outwatch/OutWatchDomSpec.scala]: Ok! I just tried in the scala REPL and it was "1.0".

Comment (mariusmuja) [reply src/test/scala/outwatch/OutWatchDomSpec.scala]: Scala REPL is JVM.

Comment (fdietze) [reply src/test/scala/outwatch/OutWatchDomSpec.scala]: I wanted to say that. 😄

**Pull request merged**